fcsparser/api.py

```diff
@@ -27,11 +27,39 @@
 
 def fromfile(file, dtype, count, *args, **kwargs):
     """Wrapper around np.fromfile to support any file-like object."""
+    
+    dtypes = dtype.split(',')
+    field_width = []
+    
+    for dt in dtypes:
+        num_bytes = int(dt[2:])
+        field_width.append(num_bytes)
+        
     try:
-        return numpy.fromfile(file, dtype=dtype, count=count, *args, **kwargs)
+        ret = numpy.fromfile(file, 
+                             dtype=",".join(['u1'] * sum(field_width)), 
+                             count=count, 
+                             *args, 
+                             **kwargs)
     except (TypeError, IOError):
-        return numpy.frombuffer(file.read(count * numpy.dtype(dtype).itemsize),
-                                dtype=dtype, count=count, *args, **kwargs)
+        ret = numpy.frombuffer(file.read(count * sum(field_width)),
+                               dtype=",".join(['u1'] * sum(field_width)), 
+                               count=count, 
+                               *args, 
+                               **kwargs)
+
+    ret = ret.view('u1').reshape((count, sum(field_width)))
+    ret_dtypes = []
+    for field, dt in enumerate(dtypes):
+        dtype_type = dt[1]
+        dtype_endian = dt[0]
+        num_bytes = int(dt[2:])
+        while num_bytes & (num_bytes - 1) != 0:
```

Comment:
The while loop block and the zero masking logic below are *very* difficult to understand

Could you add documentation to explain why this is done the way it is?

Also is it possible to entirely kill the while loop (would modulo division by 3 work?)?


```
def f(x):
    i = 0 
    x & (x-1) == 0
    while x & (x-1) != 0:
        x = x + 1
        i += 1
    return i
for x in range(1, 20):
    print x, f(x)
```

This is what the output looks like... and it doesn't seem quite right (but I don't fully understand what the code is supposed to do)

```
1 0                                                                
2 0                                                                
3 1                                                                
4 0                                                                
5 3                                                                
6 2                                                                
7 1                                                                
8 0                                                                
9 7                                                                
10 6                                                               
11 5                                                               
12 4                                                               
13 3                                                               
14 2                                                               
15 1                                                               
16 0                                                               
17 15                                                              
18 14                                                              
19 13 
```